tests/TESTINS5.MLC

```diff
@@ -341,6 +374,221 @@ BIC_OK2  EQU    *
          IILL   R3,4
          CLG    R3,=AL2(1,2,3,4)
          RT1    CC0
+* MGRK #300
+         LMG   R5,R6,=FL8'2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'123456,10000000'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'1234560000000'
+         RT1    CC0 
+* MS
+         L      R1,=F'2'
+         MS    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MS    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MS    R1,=F'-3'
+         CL     R1,=F'6'
+         RT1  CC0
+* MSY
+         L      R1,=F'2'
+         MSY    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MSY    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MSY    R1,=F'-3'
+          CL     R1,=F'6'
+          RT1  CC0
+* MSGRKC
+         LMG   R5,R6,=FL8'2,3'
+         SR    R0,R0
+         MSGRKC  R2,R5,R6
+         RT1    CC2
+         CLG    R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         MSGRKC  R2,R5,R6
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         MSGRKC  R2,R5,R6
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         MSGRKC  R2,R5,R6
+         CLG   R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=XL8'7FFFFFFFFFFFFFFF,2'
+         MSGRKC  R2,R5,R6
+          RT1    CC3  
```

Comment:
Even though overflow occurs, GR2 is still set to a value. You might want to add
```
         CLG   R2,=FL8'-2'
         RT1   CC0
```
You should, of course, compute the expected value yourself. I did it in Eclipse. Code snippet:
```
bi = new BigInteger("9223372036854775807");
        ba = bi.toByteArray();
        System.out.println("bi "+bi.toString());
        printByteArray("ba", ba);
        
        bi = bi.multiply(new BigInteger("2"));
        ba = bi.toByteArray();
        System.out.println("bi * 2 = "+bi.toString());
        printByteArray("ba", ba);
        
        ba[0] = (byte)0xff;
        bi = new BigInteger(ba);
        ba = bi.toByteArray();
        System.out.println("bi re-build = "+bi.toString());
        printByteArray("ba", ba);
```
and resulting output
```
bi 9223372036854775807
ba: 7f ff ff ff ff ff ff ff 
bi * 2 = 18446744073709551614
ba: 00 ff ff ff ff ff ff ff fe 
bi re-build = -2
ba: fe 
```
Here's the printByteArray() submodule:
```
    static void printByteArray(String name, byte[] b)
    {
        System.out.print(name+": ");
        for (int i = 0; i < b.length; i++)
            System.out.printf("%02x ",b[i]);
        System.out.println();
    }
```

Reply:
For completeness, you might add MSGRKC tests for 2 * 0, 0 * 2, (-2) * 0, 0 * (-2), and 0 * 0.